### DISCUSSION TITLE
Select control to automatically add validation for options

### DISCUSSION THREAD

**Discussion by titantwentyone** [filamentphp/filament#1902] [Ideas]
Currently, when we create a Select component, we have to manually add in validation for the options provided:

```
Select::make('gender')
    ->options([
        'male' => 'Male',
        'female' => 'Female',
        'notdisclosed' => 'Not Disclosed'
    ])
    ->rules(['in:male,female,notdisclosed'])
```
It would be good if the rule was automatically added.

I may have a stab at this myself if I get time if no objections/thoughts - just adding here as a reminder.

**Comment by ryangjchandler** [comment]
Yeah, I agree. This is a sensible default - if you don't have any time, let me know and I'll open a PR.

**Reply by titantwentyone** [reply]
Not sure when I'll get time - bit snowed under this week - so please fire away. Thanks Ryan.

**Reply by titantwentyone** [reply]
@ryangjchandler just added a [new discussion](#1911) which is loosly related to what we are discussing here. Not sure what impact this would have on a PR for this item but probably worth taking a look.

**Reply by titantwentyone** [reply]
Dan kindly answered my question above. Just another thought - the `BelongsToSelect` component could also do with this. Adding here for reference. Happy to open a new discussion/issue if needed. Just happen to be reworking an app and coming across these types of things.
